src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java

```diff
@@ -640,7 +662,18 @@ private void handleFinished()
                 final long newApplicationId = supplyStreamId.getAsLong();
                 final long applicationRef = route.targetRef();
 
-                doTlsBegin(applicationTarget, newApplicationId, authorization, applicationRef, newCorrelationId, tlsHostname);
+                String applicationProtocol = tlsEngine.getApplicationProtocol();
+                if (applicationProtocol == null || applicationProtocol.isEmpty())
+                {
+                    applicationProtocol = tlsRouteEx.applicationProtocol().asString();
+                }
+                if (applicationProtocol == null || applicationProtocol.isEmpty())
+                {
+                    applicationProtocol = "";
+                }
```

Comment:
This code implies that the `TlsBeginEx` does not always contain the application protocol that was actually negotiated  during the TLS handshake, which doesn't seem right.

Special case for empty string doesn't look right either.

What breaks if lines `666-673` are removed?

Suggest also moving line `665` to `658` and renaming to `tlsApplicationProtocol` alongside `tlsHostname`.

Reply:
Not applicable anymore